lib/roku-develop.coffee

```diff
@@ -245,6 +248,41 @@ module.exports        = RokuDevelop =
   toggle: ->
     if @panel.isVisible() then @panel.hide() else @panel.show()
 
+
+  #
+  # Invoked when the roku-develop:switch-files command is issued
+  #
+  switch: ->
+    editor = atom.workspace.getActivePaneItem()
+    panes = atom.workspace.getPanes()
+    if editor and editor.buffer
+      file = editor.buffer.file
+      if file
+        switchMade = false
+        filename = file.path.substring(file.path.lastIndexOf("/") + 1)
+        fileplain = filename.replace(/\.[^/.]+$/, "")
+        folderpath = file.path.substring(0, file.path.lastIndexOf("/"))
+        fs.readdir folderpath, {}, (err, files) =>
+          for xfile in files
+            if xfile != filename
+              if xfile.substring(0, fileplain.length) == fileplain and xfile.indexOf(".") > -1
+                xfiletype = xfile.substring(xfile.lastIndexOf(".") + 1)
+                if xfiletype == "xml" or xfiletype == "brs"
+                  if panes
+                    for pane in panes
+                      for tab in pane.getItems()
+                        if tab.getTitle() == xfile
+                          pane.activateItem(tab)
+                          pane.activate()
+                          switchMade = true
+                        return if switchMade
+                      return if switchMade
+                  if switchMade == false
+                    atom.workspace.open(folderpath + "/" + xfile)
+                    switchMade = true
+            return if switchMade
+
+
```

Comment:
I think we can simplify the switch() code by removing the 3 nested loops, replacing them with Atom API calls to get pane items from the URI; by removing all the string manipulation code, instead using the node.js path module parse/format methods; and by eliminating access to the Atom API properties which are not documented (editor.buffer, editor.buffer.file, and file.path), instead using documented API calls.

I think it should work if we replace this:

https://github.com/belltown/roku-develop/blob/8b99c986a23559cd82608ef9f759bded3e65cae4/lib/roku-develop.coffee#L255-L283

with this:

```suggestion
  switch: ->
    editor = atom.workspace.getActiveTextEditor()
    editorPath = editor?.getPath()
    if editorPath
      editorPathParsed = path.parse(editorPath)
      editorPathParsed.ext =
        if editorPathParsed.ext is '.brs' then '.xml'
        else if editorPathParsed.ext is '.xml' then '.brs'
        else ''
      if editorPathParsed.ext
        editorPathParsed.base = ''  # path.format() ignores ext if base present
        switchPath = path.format(editorPathParsed)
        pane = atom.workspace.paneForURI(switchPath)
        if not pane?.activateItemForURI(switchPath)
          fileObj = new File(switchPath)
          fileObj?.exists().then((fileExists) ->
            atom.workspace.open(switchPath) if fileExists)
```

Reply:
You're right, that's about a million times more elegant.